### PR TITLE
Internal: Create progress bar component [ED-15439]

### PR DESCRIPTION
**Pull request by elchugreeva**

### CODE REVIEW

File: modules/checklist/assets/js/components/checklist-wrapper.js [hunk 1]
```diff
@@ -20,3 +20,7 @@ const ChecklistWrapper = () => {
 };
 
 export default ChecklistWrapper;
+
+ChecklistWrapper.propTypes = {
+	steps: PropTypes.object.isRequired,
```
Comment (ronenelementor) [inline modules/checklist/assets/js/components/checklist-wrapper.js:25]:
```suggestion
	steps: PropTypes.array.isRequired,
```

File: modules/checklist/assets/js/components/progress.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import PropTypes from 'prop-types';
+
+const Progress = ( { steps } ) => {
+	const [ value, setValue ] = useState( 0 );
+
+	useEffect( () => {
+		const totalSteps = steps.length,
+			doneSteps = steps.filter( ( i ) => i.done ).length;
+
+		setValue( doneSteps / totalSteps * 100 );
```
Comment (ronenelementor) [inline modules/checklist/assets/js/components/progress.js:12]:
```suggestion
	const progress = getProgressPercentage();

	const getProgressPercentage = () => {
		return steps.map( ( { is_completed } ) => is_completed ) * 100 / steps.length;
	};	
```

Comment (ronenelementor) [reply modules/checklist/assets/js/components/progress.js:12]: 1. value isn't very suggestive in my opinion
2. I don't see any reason for using useState and useEffect

File: modules/checklist/assets/js/data/steps.js [hunk 3]
```diff
@@ -6,5 +6,24 @@ export const steps = [
...
+		done: true,
+	},
+	{
+		id: 'add-header',
+		title: 'Create your first 3 pages',
+		imagePath: 'https://assets.elementor.com/checklist/v1/images/checklist-step-4.jpg',
+		description: 'Jumpstart your creation with professional designs form the Template Library or start from scratch.',
+		link: 'https://elementor.com/help/create-new-page/',
+		CTA: 'Create a new page',
+		done: false,
```
Comment (ronenelementor) [inline modules/checklist/assets/js/data/steps.js:18]:
```suggestion
		is_completed: false,
```

**Review by ronenboxer — APPROVED** [review]

**Review by davseve — APPROVED** [review]

**Pull request merged**